src/dispatcher/actions.ts

```diff
@@ -351,4 +351,19 @@ export enum Action {
      * Fired when we want to view a thread, either a new one or an existing one
      */
     ShowThread = "show_thread",
+
+    /**
+     * Fired when requesting to ask to join a room.
+     */
+    AskToJoin = "ask_to_join",
+
+    /**
+     * Fired when requesting to submit an ask to join a room. Use with a SubmitAskToJoinPayload.
+     */
+    SubmitAskToJoin = "submit_ask_to_join",
```

Comment:
What difference is there between the two? 

`AskToJoin` is ambiguous, "fired when request to ask to join a room" implies that a request has been submitted. But the second action below contradicts that.

Could we clear that up?

Reply:
For the "guess knock" (see description) I need away to bring the form up for the user. Maybe the naming is not the best. I changed it to `PromptAskToJoin`. I hope it make the difference clearer.